flytekit/types/structured/structured_dataset.py

```diff
@@ -625,6 +623,24 @@ def t2(in_a: Annotated[StructuredDataset, kwtypes(col_b=float)]): ...
         # we should do the opening/downloading and whatever else it might entail right now. No iteration option here.
         return self.open_as(ctx, lv.scalar.structured_dataset, df_type=expected_python_type, updated_metadata=metad)
 
+    def to_html(self, ctx: FlyteContext, python_val: typing.Any, expected_python_type: Type[T]) -> str:
+        if isinstance(python_val, StructuredDataset):
+            if python_val.dataframe is not None:
+                df = python_val.dataframe
+            else:
+                df = typing.cast(StructuredDataset, python_val).open(pd.DataFrame).all()
```

Comment:
This will have the downside of always downloading the data right?

Reply:
Great catch. I think the default should only put column names or something

Reply:
That makes sense, updated it to render column names and type only, thanks.